src/main/java/module/TicTacToeBoard.java

```diff
@@ -0,0 +1,43 @@
+package main.java.module;
+
+import java.util.HashMap;
+import java.util.List;
+public class TicTacToeBoard {
+    int row;
```

Comment:
rows, cols

Reply:
Similarly for methods

Reply:
Sorry, what is the purpose of this row, col? I thought it is being used to determine the number of rows and columns but i don't think you're using it anywhere. If not using then remove

Reply:
It is used initially to set the size of the board

Reply:
That's being done through the parameters I guess and not this.

Reply:
inside Service class initialise board method I am setting the size for the board

Reply:
But that is not setting these row and col properties. It is just being passed as parameters using which you are initializing the board attribute. row and col are not being updated anywhere.

Reply:
Got it. Name these numRows and numCols then and do not set it from any other class. Your initboard method should be setting it. Right now, when your board is getting initialized these values should also get set at the same time. If the other classes follow setting it then your board breaks.

Reply:
Okay, I will rename it, but what do you mean by "board breaks"?

Reply:
I meant if other classes forget setting it then your board won't work. So that should be done during initialization.

Always consider that your class would be used by someone else and they might forget calling some methods. Which public functions would you want to expose to them to create some abstraction.